core/src/main/java/org/jdbi/v3/core/StatementContext.java

```diff
@@ -155,6 +157,13 @@ public Object getAttribute(String key)
         return config.collectorRegistry.findCollectorFor(type);
     }
 
+    public <C extends ExtensionConfig<C>> C findConfiguration(Class<C> configType)
```

Comment:
I think we should just have `getConfig(Class<T extends ExtensionConfig>): T` and only match by exact class. Might make sense to turn ExtensionConfig into an abstract class too.

I've thought back on how this class came about in the extensions PR, and had to acknowledge that this idea of configuration by instanceof was speculative. We don't have any concrete reason to hold to it.